src/base/emu-i386/simx86/codegen-x86.c

```diff
@@ -3332,11 +3332,19 @@ static unsigned Exec_x86_asm(unsigned *mem_ref, unsigned long *flg,
 		unsigned char *ecpu, unsigned char *SeqStart)
 {
 	unsigned ePC;
+	/* can't clobber ebp/rbp with frame pointers; save/restore
+	   via variable instead of push/pop due to potential issues with
+	   x86-64 stack alignment and red zone */
```

Comment:
But we explicitly disable redzone in
makefile and use force_align_arg_pointer
in cpatch.h. Is this not enough?

Reply:
Yes I missed that. Probably simpler to just add `-fomit-frame-pointer` next to `-mno-red-zone` and allow ebp to be clobbered.

Reply:
Also try `__attribute__((optimize("omit-frame-pointer")))`.
Maybe the same treatment can also
be applied to redzones to narrow the
scope of the hack.